src/inmanta/agent/in_process_executor.py

```diff
@@ -263,97 +262,113 @@ async def execute(
-                                )
-                            except Exception as e:
-                                ctx.exception(
-                                    "Exception during dryrun for %(resource_id)s (exception: %(exception)s",
-                                    resource_id=resource.rvid,
-                                    exception=str(e),
-                                )
-                                changes = ctx.changes
-                                if changes is None:
-                                    changes = {}
-                                changes["handler"] = AttributeStateChange(current="FAILED", desired="Handler failed")
-                                await self.client.dryrun_update(
-                                    tid=self.environment, id=dry_run_id, resource=resource_id, changes=changes
-                                )
-
-                    except Exception:
-                        ctx.exception("Unable to process resource for dryrun.")
-                        changes = {}
-                        changes["handler"] = AttributeStateChange(current="FAILED", desired="Resource Deserialization Failed")
-                        await self.client.dryrun_update(
-                            tid=self.environment, id=dry_run_id, resource=resource_id, changes=changes
-                        )
-                    finally:
-                        if provider is not None:
-                            provider.close()
-
-                        finished = datetime.datetime.now().astimezone()
-                        await self.client.resource_action_update(
-                            tid=self.environment,
-                            resource_ids=[resource_id],
-                            action_id=ctx.action_id,
-                            action=const.ResourceAction.dryrun,
-                            started=started,
-                            finished=finished,
-                            messages=ctx.logs,
-                            status=const.ResourceState.dry,
-                        )
+
+                except Exception as e:
+                    # This one doesn't make sense, right? It's just a log line in the try
+                    ctx.exception(
+                        "Unable to process resource %(resource_id)s for dryrun (exception: %(exception)s",
+                        resource_id=resource.rvid,
+                        exception=str(e),
+                    )
+                    dryrun_result = DryrunResult(
+                        rvid=resource_id,
+                        dryrun_id=dry_run_id,
+                        changes={"handler": AttributeStateChange(current="FAILED", desired="Resource Deserialization Failed")},
+                        started=started,
+                        finished=datetime.datetime.now().astimezone(),
+                        messages=ctx.logs,
+                    )
```

Comment:
If I understand correctly, this is for this try:
```
               try:
                    self.logger.debug("Running dryrun for %s", resource_id)
```
Are we expecting any errors from this?

Reply:
this try is to make sure we dispose of the provider..